CLAUDE.MD

```diff
@@ -0,0 +1,173 @@
+    example: "A1:B10"
+
+  - name: param2
+    description: Description of second parameter
+    default: 1
+    example: 2
+
+formula: |
+  LET(
+    variable, expression,
+    result
+  )
+
+notes: >
+  Implementation notes and usage tips.
+```
+
+## Development Workflow
+
+### Adding a New Formula
+
+1. Create a new `.yaml` file in the root directory (e.g., `myformula.yaml`)
+2. Follow the YAML schema structure above
+3. Ensure all required fields are present and valid
+4. Run `uv run generate_readme.py` locally to:
+   - Validate the YAML file
+   - Update README.md with the new formula
+5. Commit both the `.yaml` file and the updated `README.md`
+6. Push to GitHub
+
+### Testing README Generation
+
+Before committing, test the README generation locally:
+
+```bash
+uv run generate_readme.py
+```
+
+This will:
+- Scan all `.yaml` files in the root directory
+- Validate each file against the schema
+- Generate/update `README.md` from `.readme-template.md`
+- Report any validation errors
+
+### GitHub Actions Automation
+
+The repository uses GitHub Actions to automatically regenerate the README:
+
+- **On push to main**: If any `.yaml` files or related files are modified, the workflow:
+  1. Runs `generate_readme.py`
+  2. Commits and pushes updated `README.md` if changed
+
+- **On pull requests**: If README needs regeneration, the workflow adds a comment to the PR requesting the author to regenerate locally
+
+## Important Files
+
+### generate_readme.py
+
+Python script that:
+- Discovers all `.yaml` files in the root directory
+- Validates them against the formula schema
+- Generates formula list in markdown format
+- Updates README.md by replacing content between `<!-- AUTO-GENERATED CONTENT START -->` and `<!-- AUTO-GENERATED CONTENT END -->` markers
+
+### .readme-template.md
+
+Template file for README generation. Contains:
+- Static content (project description, contributing guide, license)
+- Auto-generation markers where formula list is inserted
+
+### Validation Rules
+
+The script validates:
+- All required fields are present and non-empty
+- Field types are correct (strings, lists, dicts)
+- Parameters have required `name` and `description` fields
+- YAML syntax is valid
+
+## Common Tasks
+
+### Add a new formula
+1. Create `newformula.yaml` following the schema
+2. Run `uv run generate_readme.py` to validate and update README
+3. Commit both files
+
+### Update an existing formula
+1. Edit the `.yaml` file
+2. Increment the version number
+3. Run `uv run generate_readme.py`
+4. Commit changes
+
+### Modify README static content
+1. Edit `.readme-template.md` (not README.md directly)
+2. Run `uv run generate_readme.py` to regenerate README
+3. Commit both files
+
+## Dependencies
+
+- **Python**: 3.8 or higher
+- **uv**: Package manager (used for running scripts)
+- **PyYAML**: 6.0 or higher (automatically handled by uv inline script metadata)
+
+## Excel/Google Sheets Function Reference
```

Comment:
Google Sheets only, not Excel. This needs to be consistent across the entire repository.